src/executor/nested_loop_join.rs

```diff
@@ -77,12 +79,12 @@ impl NestedLoopJoinExecutor {
             self.join_op,
             BoundJoinOperator::LeftOuter | BoundJoinOperator::FullOuter
         ) {
-            let mut i = 0;
-            for left_row in left_rows() {
+            let left_row_num = left_rows().count();
+            for (mut i, left_row) in left_rows().enumerate() {
                 let mut matched = false;
-                for _ in right_rows() {
+                for _ in 0..right_row_num {
                     matched |= matches!(filter.get(i), Some(true));
-                    i += 1;
+                    i += left_row_num;
```

Comment:
After reading this part 3 times, I'm finally sure that's correct... Would you please add some comments on what this part is doing? For example, having something like:

```rust
// Now the filter map contains the selected rows of the cross join result `right x left`. We will need to figure out whether there's an unmatched left row, and produce the correct outer join result.
// To compute the unmatched rows, we will need to first pick a row from left (namely the i-th row). Then we check if all `filter[i + left_row_num * j]` are false. If all false, we produce a (left, NULL) row.
```

Please add such comments across L82-L93.